backend/workstation/ws_controllers.js

```diff
@@ -102,13 +103,22 @@ exports.createNewFile = async (req, res) => {
       const userSocketId = userSocketMap.get(allUserIds[user_id]);
       if (userSocketId) {
         try {
-          await rateLimiter.consume(allUserIds[user_id]);
-
-          userSocketId.emit("notify_group", {
-            message: `User ${userId} has created a file: ${newFile.fileName}`,
-          });
-        } catch (rejRes) {
-          socket.emit("Too many notifications. Please try again later.");
+          const rate_limit_response = await rateLimiter.useRateLimiter(
+            allUserIds[user_id]
+          );
+          if (rate_limit_response?.error) {
```

Comment:
Why in some cases do you explicitly check the response for an error, and in others you just rely on try/catch to handle errors?

Reply:
This is not exactly an error, but a way to check if the rateLimiter has a returned response. The error element is referencing the first part of this returned object. https://github.com/izuchukwumba/peered/pull/19/files#diff-6b4a1ae09d7a7c5390c977ebfe787abaeb2ad7aa5ca0c9ce047f7e386efa54a4:~:text=/%201000%3B-,return%20%7B,%7D%3B,-%7D . Should I use something less ambigious like isRateLimited: true?

Reply:
Yeah I'm wondering why in some cases you check for this response when calling the rate limiter, and in others you don't